cache/store.go

```diff
@@ -135,3 +136,18 @@ func (s *store) scanRange(startKey, endKey []byte) []*mvccpb.KeyValue {
 func isPrefixScan(endKey []byte) bool {
 	return len(endKey) == 1 && endKey[0] == 0
 }
+
+func validateRevisions(events []*clientv3.Event, latestRev int64) error {
+	for _, ev := range events {
+		if ev.Kv.ModRevision < latestRev {
+			return fmt.Errorf("cache: stale event batch (rev %d < latest %d)", ev.Kv.ModRevision, latestRev)
+		}
+		if ev.Kv.ModRevision == latestRev {
+			return fmt.Errorf("cache: duplicate revision batch (rev %d == latest %d)", ev.Kv.ModRevision, latestRev)
+		}
+		if ev.Kv.ModRevision > latestRev+1 {
```

Comment:
You don't have guarantee of revision continuity. Are you sure you have proper test coverage if you didn't notice that?